File: server/Simulation/Components/DamageComponent.cs
using server.ECS;
using server.Enums;

namespace server.Simulation.Components;

[Component(ComponentType = ComponentType.Damage, NetworkSync = true)]
public struct DamageComponent(NTT attackerId, float damage)
{
    public long ChangedTick = NttWorld.Tick;
    public readonly NTT Attacker = attackerId;
    public float Damage = damage;
}

File: pixiejsClient/src/app/ecs/components/DamageComponent.ts
import { Component, component, serverField } from "../core/Component";
import { ServerComponentType } from "../../enums/ComponentIds";
import { NTT } from "../core/NTT";

@component(ServerComponentType.Damage)
export class DamageComponent extends Component {
  // changedTick is inherited from Component base class
  @serverField(1, "guid") attacker: string;
  @serverField(2, "f32") damage: number;

  constructor(ntt: NTT, attacker: string = "", damage: number = 0) {
    super(ntt);
    this.attacker = attacker;
    this.damage = damage;
  }
}
